src/core/execute.c

```diff
@@ -1989,8 +1992,35 @@ static int build_environment(
 
         our_env[n_env++] = x;
 
-        our_env[n_env++] = NULL;
```

Comment:
the line setting the last to NULL is lost?

Reply:
yeah, it's redundant, the array is allocated via `new0()` anyway, hence is all NULL already.